sdk/src/main/java/com/google/cloud/dataflow/sdk/io/range/OffsetRangeTracker.java

```diff
@@ -76,8 +88,14 @@ public synchronized boolean tryReturnRecordAt(boolean isAtSplitPoint, long recor
               recordStart,
               lastRecordStart));
     }
+
+    if (isAtSplitPoint && lastRecordStart > -1) {
+      ++seenSplitPoints;
```

Comment:
Add a comment here, saying that we're incrementing splitPointsProcessed on behalf of the *previous* split point, because we've seen the next one (which means the previous one has been processed)